tools/distribution/src/assets/podspec.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2019-2020 Datadog, Inc.
+# -----------------------------------------------------------
+
+import os
+import re
+import time
+import random
+from src.utils import shell, read_sdk_version
+
+
+class CPPodspec:
+    """
+    The Cocoapods podspec pushed to `pod trunk`.
+    """
+
+    __name: str  # The name of the spec, e.g. `DatadogSDK`
+    __file_name: str  # The name of the spec file, e.g. `DatadogSDK.podspec`
+    __path: str  # The path to the `podspec` file
+
+    def __init__(self, name: str):
+        file_name = f'{name}.podspec'
+        print(f'⌛️ Searching for `{file_name}` in {os.getcwd()}')
+
+        file_path = f'{os.getcwd()}/{file_name}'
+        if not os.path.isfile(file_path):
+            raise Exception(f'Cannot find `{file_name}` in {os.getcwd()}')
+
+        self.__name = name
+        self.__file_name = file_name
+        self.__path = file_path
+        print(f'   → `{file_name}` found')
+
+    def __repr__(self):
+        return f'[CPPodspec: name = {self.__name}, path = {self.__path}]'
+
+    def validate(self, git_tag: str):
+        """
+        Checks the `.podspec` integrity with given `git_tag`.
+        """
+        print(f'🔎️️ Validating {self} against: {git_tag}')
+
+        # Check if spec `.version` matches the git tag name and `sdk_version`:
+        sdk_version = read_sdk_version()
+        pod_version = self.__read_pod_version()
+        if not (pod_version == git_tag and sdk_version == git_tag):
+            raise Exception(f'`sdk_version` ({sdk_version}), `pod_version` ({pod_version})'
+                            f' and git tag ({git_tag}) do not match')
+        print(f'   → `sdk_version` ({sdk_version}), `pod_version` ({pod_version})'
+              f' and git tag ({git_tag}) do match')
+
+    def publish(self, dry_run: bool):
+        """
+        Publishes the `.podspec` to pods trunk.
+        """
+        print(f'📦 Publishing {self} ({self.__read_pod_version()})')
+
+        # Because some of our pods depend on others and due to https://github.com/CocoaPods/CocoaPods/issues/9497
+        # we need to retry `pod trunk push` until it succeeds. The maximum number of attempts is 100 (arbitrary),
+        # but this is also limited by the CI job timeout.
+        retry_time = 30  # seconds
+        attempt = 0
+        while attempt < 100:
+            attempt += 1
+            try:
+                if attempt == 1:
+                    shell(f'pod repo update', skip=dry_run)
+                    shell(f'pod spec lint --allow-warnings {self.__file_name}', skip=dry_run)
+                    shell(f'pod trunk push --synchronous --allow-warnings {self.__file_name}', skip=dry_run)
+                else:
+                    shell(f'pod repo update --silent', skip=dry_run)
+                    shell(f'pod spec lint  --silent --allow-warnings {self.__file_name}', skip=dry_run)
+                    shell(f'pod trunk push --allow-warnings {self.__file_name}', skip=dry_run)
```

Comment:
I *think* `pod trunk push` performs the same validation as  `pod spec lint`.

Reply:
Well, actually we can have both by adding a bit more complexity to this tool and capturing `shell('')` output along with its result. Then we will control which output to print in the log. So this is a third option to consider 👈 .